File: LeetCode/0796. Rotate String/solution.go
/**
 * 47 / 47 test cases passed.
 * Runtime: 0 ms
 * Memory Usage: 1.8 MB 
 */
 func rotateString(s string, goal string) bool {
    return len(s) == len(goal) && strings.Contains(s + s, goal)
}

File: LeetCode/0796. Rotate String/solution.rs
/**
 * 47 / 47 test cases passed.
 * Runtime: 0 ms
 * Memory Usage: 2.2 MB 
 */
impl Solution {
    pub fn rotate_string(s: String, goal: String) -> bool {
        s.len() == goal.len() && s.repeat(2).contains(&goal)
    }
}
